runtime/include/scheduler.h

```diff
@@ -254,6 +254,10 @@ scheduler_preemptive_sched(ucontext_t *interrupted_context)
 	/* Preempt executing sandbox */
 	scheduler_log_sandbox_switch(interrupted_sandbox, next);
 	sandbox_preempt(interrupted_sandbox);
+
+	// Write back global at idx 0
+	sledge_abi__wasm_globals_set_i64(0, sledge_abi__current_wasm_module_instance.abi.wasmg_0, true);
```

Comment:
I am having hard time to recall why this needed to be done during the preemptive_sched and not the cooperative_sched?

Reply:
Really good catch! I think you're right. This would be a problem when a sandbox blocks. Making the change and will run some additional tests shortly.

Reply:
Also not really sure when I'm calling the "ABI" here since we're inside of the runtime already. I'm think these should probably be `wasm_globals_set_i64` instead